warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql

```diff
@@ -0,0 +1,114 @@
+WITH payments_rides AS (
+    SELECT * FROM {{ ref('payments_rides') }}
+),
+
+route_z_rides AS (
+    SELECT * FROM payments_rides
+    WHERE route_id = 'Route Z'
+),
+
+null_rides AS (
+    SELECT * FROM payments_rides
+    WHERE route_id IS NULL
+),
+
+payments_tests_monthly_date_spine AS (
+    SELECT * FROM {{ ref('payments_tests_monthly_date_spine') }}
+),
+
+count_route_z_rides AS (
+    SELECT
+
+        participant_id,
+        month_start,
+        COUNT(*) AS n_route_z_rides
+
+    FROM payments_tests_monthly_date_spine
+    INNER JOIN route_z_rides
+        USING (participant_id) WHERE transaction_date_pacific >= month_start AND transaction_date_pacific <= month_end
+    GROUP BY month_start, participant_id
+),
+
+count_null_rides AS (
+    SELECT
+
+        participant_id,
+        month_start,
+        COUNT(*) AS n_null_rides
+
+    FROM payments_tests_monthly_date_spine
+    INNER JOIN null_rides
+        USING (participant_id) WHERE transaction_date_pacific >= month_start AND transaction_date_pacific <= month_end
+    GROUP BY month_start, participant_id
+),
+
+count_all_rides AS (
+    SELECT
+
+        participant_id,
+        month_start,
+        COUNT(*) AS n_all_rides
+
+    FROM payments_tests_monthly_date_spine
+    INNER JOIN payments_rides
+        USING (participant_id) WHERE transaction_date_pacific >= month_start AND transaction_date_pacific <= month_end
+    GROUP BY month_start, participant_id
+),
+
+match_date_spine AS (
+    SELECT
+
+        date_spine.participant_id,
+        date_spine.month_start,
+        COALESCE(z_rides.n_route_z_rides, 0) AS n_route_z_rides,
+        COALESCE(null_rides.n_null_rides, 0) AS n_null_rides,
+        (COALESCE(z_rides.n_route_z_rides, 0) + COALESCE(null_rides.n_null_rides, 0)) AS total_mislabeled_rides,
+        COALESCE(all_rides.n_all_rides, 0) AS n_all_rides
+
+    FROM payments_tests_monthly_date_spine AS date_spine
+    LEFT JOIN count_all_rides AS all_rides
+        USING (participant_id, month_start)
+    LEFT JOIN count_route_z_rides AS z_rides
+        USING (participant_id, month_start)
+    LEFT JOIN count_null_rides AS null_rides
```

Comment:
You should be able to do these all at once in the aggregation; I'd recommend doing it for practice. You can use conditions to count different columns; for example [COUNTIF](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#countif) to check for specific values. `count(column)` already only counts not-null rows for that column; `count(0)/count(*)` counts the row regardless.